### DISCUSSION TITLE
Is it possible to use properties inside <head>?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#862] [Q&A]
Hi

I would like to use a property for elements inside `<head>` tag.

Till now, I used mavo inside the `<body>` tag only.

Is it possible (and fine to do) and if so, how would be the best way to do it, considering 2 use cases:

1. property on element `<title>`

is it ok to use something like: 
`<title property="title">my title</title>`

2. property on element `<meta>` which already have an attribute 'property' (!)

- -> How to use it with 'stable' mavo version?

`<meta property="og:title" content="Description of my page">`

- -> or is it correct to use it like this with 'dev' mavo version?

`<meta property="og:title" content="Description of my page" mv-attr-content="THE NEW DESCRIPTION">`

**Comment by DmitrySharabin** [comment]
Hey there!

Sorry for not replying for so long: Lea is at [the conference](https://cssday.nl/2022) right now, and I was a bit swamped.

>I would like to use a property for elements inside <head> tag. Is it possible...

You can make the whole page an app by adding `mv-app` to the `<html>` element. [This demo](https://mavo.io/demos/countdown/) should help.

> is it ok to use something like:
><title property="title">my title</title>

You can definitely use the value of that property inside your app. But how are you planning to edit it? Do you expect the `<title>` element become visible in edit mode?

> `<meta property="og:title" content="Description of my page">`

This is an interesting case. I checked it with the [Mavo Inspector](https://github.com/mavoweb/inspector) and found that Mavo finds the `og:title` property. However, we can't use it in expressions—Mavo can't handle it because of `:` (it's invalid in [property names](https://mavo.io/docs/properties#property-name-rules)).

Btw, `mv-attr-*` is not in the dev version yet (it's still a [PR](https://github.com/mavoweb/mavo/pull/833), and we are working on it). If we have it in the future it would work only with expressions. So, one would need to write something like this to make it work: `mv-attr-content="[expression]"`.

If the only thing you need is to be able to set `<title>` (and/or other page metadata), I would suggest using expressions for this, and defining the corresponding properties inside `<body>`, e.g., via `<meta>` elements, like so:

```html
<html mv-app="appName">
  <head>
    <title mv-value="page_title">Default Title</title>
    ...
  </head>
  <body>
    <meta property="page_title" content="Page Title" />
  </body>
</html>
``` 

I used `mv-value` here so that the user can see `Default Title` instead of `[page_title]` [while Mavo is being loaded](https://mavo.io/docs/expressions#providing-fallbacks-the-mv-value-attribute).

Hope it helps!

**Reply by DmitrySharabin** [reply]
And one more thing. Please consider [this bug](https://github.com/mavoweb/mavo/issues/800) we are aware of and are planning to fix in the upcoming release.

**Reply by GalinhaLX** [reply]
Hi @DmitrySharabin 

Thanks for your reply. All is fine. 

About : 

> You can definitely use the value of that property inside your app. But how are you planning to edit it? Do you expect the `<title>` element become visible in edit mode?

I'm thinking about another property, linked to this one 
That's why I needed to solve  [#854 ](https://github.com/mavoweb/mavo/issues/854) of having same property multiple times inside same app, or linking them trought different apps


or maybe with a Server Side process (my next questions will come on how to set Mavo SSR :) )

**Reply by DmitrySharabin** [reply]
> (my next questions will come on how to set Mavo SSR :) )

:) Have you already tried [Mavo SSR](https://github.com/mavoweb/mavo-ssr)?

**Reply by GalinhaLX** [reply]
I've been trying to install Mavo SSR without success :( on [replit](https://replit.com)

I also have questions about express.js, because, plugin documentation,  mentions not to use it in production
Should I use something like NginX instead?

I'm also trying to understand, in my case, where I'll probably also add a PHP server on node.js (yes...), what will be the process of rendering: if Mavo SSR processces before PHP

Maybe better to open a new discussion connected to this topic

**Reply by DmitrySharabin** [reply]
> Maybe better to open a new discussion connected to this topic

Yup. And I would mention the Mavo SSR author in it to get info straight from the horse's mouth.
